mennekes/amtronecudiscovery.cpp

```diff
@@ -0,0 +1,155 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+*
+* Copyright 2013 - 2022, nymea GmbH
+* Contact: contact@nymea.io
+*
+* This file is part of nymea.
+* This project including source code and documentation is protected by
+* copyright law, and remains the property of nymea GmbH. All rights, including
+* reproduction, publication, editing and translation, are reserved. The use of
+* this project is subject to the terms of a license agreement to be concluded
+* with nymea GmbH in accordance with the terms of use of nymea GmbH, available
+* under https://nymea.io/license
+*
+* GNU Lesser General Public License Usage
+* Alternatively, this project may be redistributed and/or modified under the
+* terms of the GNU Lesser General Public License as published by the Free
+* Software Foundation; version 3. This project is distributed in the hope that
+* it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+* warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this project. If not, see <https://www.gnu.org/licenses/>.
+*
+* For any further details and any questions please contact us under
+* contact@nymea.io or see our FAQ/Licensing Information on
+* https://nymea.io/license/faq
+*
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include "amtronecudiscovery.h"
+#include "extern-plugininfo.h"
+
+AmtronECUDiscovery::AmtronECUDiscovery(NetworkDeviceDiscovery *networkDeviceDiscovery, QObject *parent) :
+    QObject{parent},
+    m_networkDeviceDiscovery{networkDeviceDiscovery}
+{
+    m_gracePeriodTimer.setSingleShot(true);
+    m_gracePeriodTimer.setInterval(3000);
+    connect(&m_gracePeriodTimer, &QTimer::timeout, this, [this](){
+        qCDebug(dcMennekes()) << "Discovery: Grace period timer triggered.";
+        finishDiscovery();
+    });
+}
+
+void AmtronECUDiscovery::startDiscovery()
+{
+    qCInfo(dcMennekes()) << "Discovery: Searching for AMTRON wallboxes in the network...";
+    NetworkDeviceDiscoveryReply *discoveryReply = m_networkDeviceDiscovery->discover();
+
+    // Check any already discovered infos..
```

Comment:
Please note: once this branch has landed (https://github.com/nymea/nymea/pull/576), this check is not required any more. Also please make sure then that the `NetworkDeviceDiscoveryReply` will be deleted once not needed any more.

    connect(discoveryReply, &NetworkDeviceDiscoveryReply::finished,  discoveryReply, &NetworkDeviceDiscoveryReply::deleteLater);